<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.5.1-0-g916b5db  Build Version: 1.5.7152.31018
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="kuka_robot">
  <link
    name="dummy_link">
    <pose>0 0 0 0 0 0</pose> 
  </link>
  <joint
    name="dummy_joint" type="fixed">
    <parent
      link="dummy_link" />
    <child
      link="base_link" />
  </joint>
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="0.00985839781372333 0.00983542590276781 -0.0196483503869256"
        rpy="0 0 0" />
      <mass
        value="45902.4076767524" />
      <inertia
        ixx="3375649.41135737"
        ixy="4.45076819362044"
        ixz="-144.655656348644"
        iyy="3375650.5024895"
        iyz="-143.950378489077"
        izz="6751050.82874779" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="link_1">
    <inertial>
      <origin
        xyz="-0.0407280954002678 -0.000551864939439428 0.256465716139208"
        rpy="0 0 0" />
      <mass
        value="11.768863008459" />
      <inertia
        ixx="0.114925138533639"
        ixy="-6.79265775359606E-05"
        ixz="0.0192682280799106"
        iyy="0.111940655335203"
        iyz="0.000311969912430666"
        izz="0.0395357850383032" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_1"
    type="revolute">
    <origin
      xyz="0 0 0"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="link_1" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="100"
      velocity="10" />
  </joint>
  <link
    name="link_2">
    <inertial>
      <origin
        xyz="-6.11968305308519E-07 -0.058001352318151 0.0424343048420221"
        rpy="0 0 0" />
      <mass
        value="10.7897563285331" />
      <inertia
        ixx="0.0830967489950308"
        ixy="-2.07962161404763E-06"
        ixz="-9.61769693825744E-07"
        iyy="0.0338827495499679"
        iyz="-0.0104752368447231"
        izz="0.0818331866977159" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_2"
    type="revolute">
    <origin
      xyz="0 0 0.32871"
      rpy="-1.5707963267949 -0.0100264683973456 -1.5707963267949" />
    <parent
      link="link_1" />
    <child
      link="link_2" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="1000"
      velocity="100" />
  </joint>
  <link
    name="link_3">
    <inertial>
      <origin
        xyz="-0.00721704489989539 0.0478236490038899 0.343253344083378"
        rpy="0 0 0" />
      <mass
        value="8.79743643489921" />
      <inertia
        ixx="0.0848464105560716"
        ixy="0.000571901369291471"
        ixz="0.00182122955381012"
        iyy="0.0796381518850136"
        iyz="-0.0155519817008965"
        izz="0.0256931179021889" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_3"
    type="revolute">
    <origin
      xyz="0 0 0"
      rpy="1.5707963267949 0 0.0100264683973456" />
    <parent
      link="link_2" />
    <child
      link="link_3" />
    <axis
      xyz="-0.0100263004046119 0 0.999949735386833" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="100"
      velocity="10" />
  </joint>
  <link
    name="link_4">
    <inertial>
      <origin
        xyz="2.75337436184131E-09 0.0709472400730481 0.0494472867754304"
        rpy="0 0 0" />
      <mass
        value="9.98676107364807" />
      <inertia
        ixx="0.0716960860200632"
        ixy="8.50602075343622E-08"
        ixz="-1.97275496651491E-07"
        iyy="0.0305436143091624"
        iyz="0.00763504269508548"
        izz="0.0641444406394451" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_4.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_4.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_4"
    type="revolute">
    <origin
      xyz="-0.00422006984030032 0 0.420878843624318"
      rpy="1.57000742485438 0.0127637059614664 0.0787442209504198" />
    <parent
      link="link_3" />
    <child
      link="link_4" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="100"
      velocity="10" />
  </joint>
  <link
    name="link_5">
    <inertial>
      <origin
        xyz="-0.00529878368038217 -0.0736286201957299 0.312599628987939"
        rpy="0 0 0" />
      <mass
        value="6.21029728548918" />
      <inertia
        ixx="0.0599432722236602"
        ixy="-0.00100412111285406"
        ixz="0.00149466318428371"
        iyy="0.0429072988766953"
        iyz="0.0221887720838403"
        izz="0.0273670468077093" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_5.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_5.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_5"
    type="revolute">
    <origin
      xyz="0 0 0"
      rpy="-1.5707963267949 0 0" />
    <parent
      link="link_4" />
    <child
      link="link_5" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="100"
      velocity="10" />
  </joint>
  <link
    name="link_6">
    <inertial>
      <origin
        xyz="9.56141660984899E-06 0.0140886191360723 0.000690324602346942"
        rpy="0 0 0" />
      <mass
        value="3.34280770004171" />
      <inertia
        ixx="0.00853740824178987"
        ixy="1.573500947614E-06"
        ixz="-5.04464720441374E-07"
        iyy="0.00539320967547681"
        iyz="-0.000275409058422702"
        izz="0.00800781843699721" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_6.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/link_6.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_6"
    type="revolute">
    <origin
      xyz="0 0 0.37092"
      rpy="-1.56900596810465 -0.0670133716499376 -0.0785324325804236" />
    <parent
      link="link_5" />
    <child
      link="link_6" />
    <axis
      xyz="0 0.00179035773378448 0.999998397308308" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="100"
      velocity="10" />
  </joint>
  <link
    name="end_effector">
    <gravity>0</gravity>
    <inertial>
      <origin
        xyz="7.41806616133545E-11 -1.51179346818964E-11 0.0809873767016411"
        rpy="0 0 0" />
      <mass
        value="1.08451608142431" />
      <inertia
        ixx="0.00100977119620439"
        ixy="-2.61683453721292E-12"
        ixz="-2.50500139091525E-12"
        iyy="0.00100977120781072"
        iyz="2.85513599731064E-13"
        izz="0.00129436470277218" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/end_effector.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/end_effector.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="end_effector_joint"
    type="revolute">
    <origin
      xyz="0 0 0"
      rpy="1.56900596810465 0 0" />
    <parent
      link="link_6" />
    <child
      link="end_effector" />
    <axis
      xyz="0 0 1" />
    <limit effort="30" velocity="1.0" lower="0" upper="0" />
  </joint>
  
  <link
    name="arm_1">
    <inertial>
      <origin
        xyz="0.00450303183379489 0.0918071952823314 1.55599715456756E-06"
        rpy="0 0 0" />
      <mass
        value="0.213632888580789" />
      <inertia
        ixx="0.000863014221928903"
        ixy="0.000102410510714024"
        ixz="-1.49716069544515E-10"
        iyy="7.16888369385725E-05"
        iyz="3.5622252891431E-10"
        izz="0.000925526664743053" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/arm_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/arm_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="arm_joint_1"
    type="revolute">
    <origin
      xyz="-0.000873164992876596 -0.0408394699522909 0.102048991638737"
      rpy="1.5707963267949 0.115992101759823 -1.59217348919776" />
    <parent
      link="end_effector" />
    <child
      link="arm_1" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="100"
      velocity="10" />
  </joint>
  <link
    name="arm_2">
    <inertial>
      <origin
        xyz="0.00450285393014704 0.0918073737419027 1.57330924699384E-06"
        rpy="0 0 0" />
      <mass
        value="0.213631104583144" />
      <inertia
        ixx="0.000863014506865485"
        ixy="0.000102409520314528"
        ixz="-3.87979397524149E-10"
        iyy="7.16879770102082E-05"
        iyz="4.28899422989554E-10"
        izz="0.000925526164783794" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/arm_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/arm_2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="arm_joint_2"
    type="revolute">
    <origin
      xyz="-0.0349314359593366 0.0211759180416709 0.102048991638737"
      rpy="1.5707963267949 0.0186923364025925 2.59661671558864" />
    <parent
      link="end_effector" />
    <child
      link="arm_2" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="100"
      velocity="10" />
  </joint>
  <link
    name="arm_3">
    <inertial>
      <origin
        xyz="0.00450231732890799 0.0918072386088336 1.58567795385811E-06"
        rpy="0 0 0" />
      <mass
        value="0.213630424459881" />
      <inertia
        ixx="0.000863017892436784"
        ixy="0.000102405326537618"
        ixz="-1.08650228730959E-09"
        iyy="7.16878973968062E-05"
        iyz="2.19232299778411E-10"
        izz="0.0009255295087147" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/arm_3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://kuka_robot/meshes/arm_3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="arm_joint_3"
    type="revolute">
    <origin
      xyz="0.0358046009522075 0.0196635519106159 0.102048991638735"
      rpy="1.5707963267949 0.0868166221658283 0.502221613195431" />
    <parent
      link="end_effector" />
    <child
      link="arm_3" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="100"
      velocity="10" />
  </joint>
    <!-- Adding Transmissions -->
    
    <transmission name="trans_joint_1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_1_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint_2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_2_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint_3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_3">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_3_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint_4">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_4">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_4_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint_5">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_5">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_5_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint_6">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_6">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_6_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_end_effector_joint">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="end_effector_joint">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="end_effector_joint_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_arm_joint_1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="arm_joint_1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="arm_joint_1_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_arm_joint_2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="arm_joint_2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="arm_joint_2_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_arm_joint_3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="arm_joint_3">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="arm_joint_3_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    
    <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>/kuka_robot</robotNamespace>
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
      <legacyModeNS>true</legacyModeNS>
    </plugin>
  </gazebo>
</robot>